utils/sl/components/__init__.py

```diff
@@ -299,9 +298,9 @@ def define_components(reqs):
     retriever = GitRepoRetriever("https://github.com/spdk/spdk.git", True)
     reqs.define('spdk',
                 retriever=retriever,
-                commands=['cd dpdk; '                            \
-                          'git fetch; '                          \
-                          'git checkout origin/spdk-19.11.6',    \
```

Comment:
I think this was doing a checkout on the spdk tree, not the dpdk tree, which directly contradicts the build.config file, and at the very least means we weren't building with what we thought we were, and possibly means that we were building with versions with known vulnerabilities.

Reply:
> I think this was doing a checkout on the spdk tree, not the dpdk tree

I don't think that is correct.  If it is then this output from the build:

```
[2021-04-27T14:45:56.234Z] RUN: cd dpdk; git fetch; git checkout origin/spdk-19.11.6
```

is misleading **at best**.  That is entirely implying (to me at least) that that is being run in a single shell.

But in fact I believe it is.  The dpdk branch that is being checked out doesn't even exist in the spdk repository:

```
$ cd spdk
$ git fetch
remote: Enumerating objects: 4734, done.
remote: Counting objects: 100% (3889/3889), done.
remote: Compressing objects: 100% (844/844), done.
remote: Total 4734 (delta 3192), reused 3727 (delta 3035), pack-reused 845
Receiving objects: 100% (4734/4734), 2.11 MiB | 2.61 MiB/s, done.
Resolving deltas: 100% (3613/3613), completed with 546 local objects.
From github.com:spdk/spdk
   48aed8a57..9ed384dab  master     -> origin/master
   02f9f4552..64f1ea8ea  v21.01.x   -> origin/v21.01.x
 * [new branch]          v21.04.x   -> origin/v21.04.x
 * [new tag]             v21.04-rc1 -> v21.04-rc1
 * [new tag]             v21.01.1   -> v21.01.1
Fetching submodule dpdk
From https://github.com/spdk/dpdk
   707692e67d..eb16786836  spdk-20.11 -> origin/spdk-20.11
 * [new branch]            spdk-21.02 -> origin/spdk-21.02
Fetching submodule ocf
From https://github.com/Open-CAS/ocf
 * [new branch]      fix-freeing-discard-reqs -> origin/fix-freeing-discard-reqs
 * [new branch]      fix-io-put             -> origin/fix-io-put
 * [new branch]      fix-removing-acp-cores -> origin/fix-removing-acp-cores
   fe56d71..2d62594  master                 -> origin/master
 * [new branch]      mio_mpool              -> origin/mio_mpool
 * [new branch]      shorten-allocator-name -> origin/shorten-allocator-name
 * [new branch]      v20.12.1               -> origin/v20.12.1
 * [new branch]      v20.12.2               -> origin/v20.12.2
 * [new branch]      v20.3.4                -> origin/v20.3.4
 * [new tag]         v20.12.1               -> v20.12.1
 * [new tag]         v20.12.2               -> v20.12.2
 * [new tag]         v20.3.4                -> v20.3.4
 * [new tag]         v21.3                  -> v21.3
 * [new tag]         v20.12                 -> v20.12
$ git checkout v20.01.2
M	dpdk
HEAD is now at b2808069e SPDK 20.01.2
$ git submodule update
Submodule path 'dpdk': checked out 'ef71bfaface10cc19b75e45d3158ab71a788e3a9'
$ git checkout origin/spdk-19.11.6
error: pathspec 'origin/spdk-19.11.6' did not match any file(s) known to git
```

But it does exist in the dpdk submodule:

```
$ cd dpdk
$ git checkout origin/spdk-19.11.6
Previous HEAD position was ef71bfafac VFIO: Fixes VFIO/sysfs race condition
HEAD is now at d09cead08d build/meson: disable libraries we don't need
```

Is scons_local really ignoring errors from the commands it's running such that a checkout of an invalid reference is being ignored and the build continuing on?  That's a bug if it is.  But even if it is, surely we'd at least see the error output, wouldn't we?  And indeed no, checking, we actually see the same output as the successful checkout of the dpdk reference above:

```
[2021-04-27T14:45:56.234Z] RUN: cd dpdk; git fetch; git checkout origin/spdk-19.11.6
[2021-04-27T14:45:57.169Z] Previous HEAD position was ef71bfa... VFIO: Fixes VFIO/sysfs race condition
[2021-04-27T14:45:57.169Z] HEAD is now at d09cead... build/meson: disable libraries we don't need
```

> at the very least means we weren't building with what we thought we were, and possibly means that we were building with versions with known vulnerabilities.

Per above, I don't think this is true.

That all said, this would not have been the case for our release packages in any case.  Our dpdk RPM is not built using the above process.  It is built separately as an RPM, very carefully to use the accepted version:

https://github.com/daos-stack/dpdk/blob/0bf016e54e52ea8bdf5458a9fa6559d961ae7853/dpdk.spec#L23 with the version defined at https://github.com/daos-stack/dpdk/blob/0bf016e54e52ea8bdf5458a9fa6559d961ae7853/dpdk.spec#L11.